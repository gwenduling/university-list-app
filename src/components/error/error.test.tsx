import Error from "./error";

import { shallow } from "enzyme";

describe("Error Component", () => {
  it("should match the snapshot", () => {
    const wrapper = shallow(<Error content="" onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should trigger callback", () => {
    const mockCallBack = jest.fn();
    const wrapper = shallow(
      <Error content="" onClick={() => mockCallBack()} />
    );

    wrapper.find("button").simulate("click");
    expect(mockCallBack).toHaveBeenCalledTimes(1);
  });
});
