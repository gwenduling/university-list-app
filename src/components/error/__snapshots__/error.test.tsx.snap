// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component should match the snapshot 1`] = `
<div
  className="error-component"
>
  <div
    className="error-content"
  >
    <span>
      Failed to fetch 
      .
    </span>
    <button
      className="button"
      onClick={[Function]}
    >
      Try again
    </button>
  </div>
</div>
`;
